plugins/woocommerce/src/Admin/API/LaunchYourStore.php

```diff
@@ -133,7 +133,7 @@ public function initialize_coming_soon() {
 		$private_link     = 'no';
 		$share_key        = wp_generate_password( 32, false );
 
-		add_option( 'woocommerce_coming_soon', $coming_soon );
+		update_option( 'woocommerce_coming_soon', $coming_soon );
```

Comment:
@chihsuan Nit: Do we have a reason to change this to `update_option`?  Using `add_option` already guarantees the option is set when it doesn't exist.

Not a popular use case, but I imagine if users wanted to go through to core profiler manually - like if they wanted to change their store details or install some plugins they remember they saw once in a core profiler, it may set their existing live site to coming soon unintentionally 🤔

Reply:
If there is a reason, I think we should also at least update the comment above:

> Initializes options for coming soon. Does not override if options exist.

Reply:
@ilyasfoo Yes,  we're now initializing coming_soon option to `no` when woocommerce is installed so we need to change this to `update_option`. Please see Adrian's comment for more details: https://github.com/woocommerce/woocommerce/pull/50581#pullrequestreview-2233555292

I'm not sure if there is a better way to handle this. I'm open to suggestions.

Reply:
We can maybe check if the core profiler is completed or skipped to avoid this issue. 🤔

Reply:
@chihsuan  Ah, I see. Thanks for the context!

Adding a more realistic use case: Sometimes when going to my store via URL, occasionally I get dropped into core profiler because it was one of the top suggested history list when I type in the domain. This happened to me a few times with JN sites.

> We can maybe check if the core profiler is completed or skipped to avoid this issue. 🤔

Great suggestion! I think this works the best.

Reply:
Okay, opened a PR here https://github.com/woocommerce/woocommerce/pull/50668